frameworks/Java/armeria/src/main/java/hello/services/PostgresFortunesService.java

```diff
@@ -0,0 +1,112 @@
+package hello.services;
+
+import hello.models.Fortune;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.annotation.Get;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+
+public class PostgresFortunesService
+{
+  private static final ObjectMapper MAPPER         = new ObjectMapper();
+  private static final DateFormat   DATE_FORMATTER = new SimpleDateFormat(
+      "E, dd MMM yyyy HH:mm:ss z");
+
+  private static final String DATABASE_HOST     = "jdbc:postgresql://tfb-database:5432/hello_world";
+  private static final String DATABASE_USER     = "benchmarkdbuser";
+  private static final String DATABASE_PASSWORD = "benchmarkdbpass";
+
+  private static final String SELECT_QUERY = "SELECT * FROM fortune";
+
+  private final MustacheFactory mustacheFactory = new DefaultMustacheFactory();
+
+  @Get("/fortunes") public HttpResponse fortunes()
+  {
+    List<Fortune> fortunes = getFortunes();
+    fortunes.add(new Fortune(0, "Additional fortune added at request time."));
+    fortunes.sort(Comparator.comparing(Fortune::getMessage));
+
+    try
+    {
+      HttpHeaders headers = HttpHeaders.of(HttpStatus.OK).add(
+          HttpHeaderNames.SERVER, "armeria").add(HttpHeaderNames.DATE,
```

Comment:
Another subjective nit-picky style comment:

You repeat this logic enough times (6) that it should be extracted to a helper method.  By "this logic" I'm referring to the logic that adds the `Server` and `Date` headers specifically; those will be the same everywhere in your application whereas the status code and `Content-Type` are likely to vary between call sites.

I suggest something like this:

```java
static HttpHeaders addCommonHeaders(HttpHeaders headers) {
    return headers.add(HttpHeaderNames.SERVER, "armeria")
                  .add(HttpHeaderNames.DATE, ...);
}
```

or this:

```java
static HttpHeaders responseHeaders(HttpStatus status, MediaType contentType) {
    return HttpHeaders.of(status)
        .add(HttpHeaderNames.SERVER, "armeria")
        .add(HttpHeaderNames.DATE, ...);
        .contentType(contentType);
}
```

Reply:
Thank you for all the great feedback. I will make the updates. As for the formatting, I used IntelliJ auto-formatting after importing the settings for the wiki. Not sure why I am getting bad formatting, I might have imported the settings incorrectly up. Either way, I will correct this.

Reply:
It could also be that the settings on our wiki are wrong.  I'm not familiar with them.  I'll send you the code style settings that I use.